### PR TITLE
Add table of content feature

### PR DESCRIPTION
**Pull request by manoj9896**
### Changes Made
- Added a header in book.jsx and use the humburger Icon
- Add a key "tableOfContents" in settings.json 
- Added style in main.sass
### Any other change
- There was a minor issue in book.jsx in the <input> element 
 ```html
<input type="number" min="1" max={pages.length} 
    onChange={(e)=>{
         if(e.target.value <= 0 || e.target.value > pages.length){
            e.target.classList.add('error')
          }else{
            e.target.classList.remove('error')
            setCurrent(e.target.value)
          }
 }} value={current} style={{width: '50px'}}></input>```

- e.target.value is string here and we were setting the string value to setCurrent
- so I used the parseInt method here before assigning it to setCurrent.

https://github.com/Bashamega/ebookCraft/assets/51627080/ade29c6f-5dec-42b7-943f-1b64504daab5

### CODE REVIEW

**Review by Bashamega — CHANGES_REQUESTED** [review]
Make it by default hidden and also make the name the page name by getting the title

**Review by Bashamega — CHANGES_REQUESTED** [review]
Also make the hamburger in the footer

**Comment by manoj9896** [comment]
> Make it by default hidden and also make the name the page name by getting the title
@Bashamega can you rephrase what you mean by **"make the name the page name by getting the title"**

**Comment by Bashamega** [comment]
> > Make it by default hidden and also make the name the page name by getting the title
> > @Bashamega can you rephrase what you mean by **"make the name the page name by getting the title"**

I mean get the title tag in the html document
